src/widget/wtracktableview.cpp

```diff
@@ -844,6 +844,44 @@ void WTrackTableView::keyPressEvent(QKeyEvent* event) {
     }
 }
 
+void WTrackTableView::resizeEvent(QResizeEvent* event) {
+    // When the tracks view shrinks in height, e.g. when other skin regions expand,
+    // and if the row was visible before resizing, scroll to it afterwards.
+
+    // these heights are the actual inner region without header, scrollbars and padding
+    int oldHeight = event->oldSize().height();
+    int newHeight = event->size().height();
+    bool maybeScrollTo = newHeight < oldHeight;
+
+    QModelIndex currIndex;
+    int rHeight = rowHeight(0);
+    if (maybeScrollTo) {
+        currIndex = currentIndex();
+        if (currIndex.isValid() && rHeight > 0) {
+            // y-pos of the top edge, negative value means above viewport boundary
+            int posInView = rowViewportPosition(currIndex.row());
+            // check if the row is visible
+            // don't use viewport()->height() because that may already have changed (??)
+            maybeScrollTo = posInView > 0 && posInView - rHeight < oldHeight;
+        } else {
+            maybeScrollTo = false;
+        }
+    }
+
+    QTableView::resizeEvent(event);
+
+    if (maybeScrollTo && currIndex.isValid()) {
```

Comment:
```suggestion
    if (maybeScrollTo) {
```
maybeScrollTo can only be true if currIndex is valid (checked above).